typing/env.mli

```diff
@@ -204,6 +204,14 @@ type shared_context =
   | Module
   | Probe
 
+type lock =
```

Comment:
As disucssed, only expose abstract `type locks`.

Reply:
In the end I expose abstract `type lock`, but the user code still know `locks` is `lock list` because they need to be able to construct an empty lock list.